lib/pbench/server/api/resources/users_api.py

```diff
@@ -340,26 +340,35 @@ def get(self, target_username):
         """
         try:
             target_user = self.get_valid_target_user(target_username)
-            if target_user:
-                response_object = target_user.get_json()
-                return make_response(jsonify(response_object), HTTPStatus.OK)
-            else:
-                self.logger.warning(
-                    "User {} is not authorized to get user {}.",
-                    self.auth.token_auth.current_user().username,
-                    target_username,
-                )
-                abort(
-                    HTTPStatus.FORBIDDEN,
-                    message=f"Not authorized to access information about user {target_username}",
-                )
         except Exception:
             self.logger.exception(
                 "Exception occurred in GET request from user {!r} while validating target user {!r}",
                 self.auth.token_auth.current_user().username,
                 target_username,
             )
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        if target_user:
+            response_object = target_user.get_json()
+            return make_response(jsonify(response_object), HTTPStatus.OK)
+        elif self.auth.token_auth.current_user().is_admin():
+            self.logger.warning(
+                "User {} performed get operation but user {} is not found.",
+                self.auth.token_auth.current_user().username,
+                target_username,
+            )
+            abort(
+                HTTPStatus.NOT_FOUND, message=f"User {target_username} not found",
+            )
+        else:
+            self.logger.warning(
+                "User {} is not authorized to get user {}.",
```

Comment:
I'm not crazy about this message. We're not technically confirming that the user name exists, but the message implies that it does, and I'd prefer we phrase this differently. I also don't think it's necessary to identify the user encoded in the authorization token. (E.g., if they somehow stole a valid auth token, let's not tell them the username.)

Reply:
But this is a log message not the abort message so I think its okay

Reply:
I'm with Nikhil:  since this goes in the server log and not back to the user, I don't think we're leaking any secrets.  (Anyone with direct access to the log also, presumably, has access to the user database itself....)  That said, I'm not opposed to rephrasing the message to make it more accurate, but the scenario here is that a non-admin user is requesting information about someone else (who does exist), and the message seems like it captures that.